core/src/main/scala/akka/kafka/internal/DefaultProducerStage.scala

```diff
@@ -28,14 +29,13 @@ import scala.util.{Failure, Success, Try}
  */
 @InternalApi
 private[kafka] class DefaultProducerStage[K, V, P, IN <: Envelope[K, V, P], OUT <: Results[K, V, P]](
-    val closeTimeout: FiniteDuration,
-    val closeProducerOnStop: Boolean,
-    val producerProvider: ExecutionContext => Future[Producer[K, V]]
+    val settings: ProducerSettings[K, V],
+    val closeProducerOnStop: Boolean // TODO: when is this ever `False`?  is it just for testing?
```

Comment:
It is false if you share the producer. You may need to add that to the settings if those are to be used to share the producer.

Reply:
Thanks.